documentation/decisions/infra/0010-production-networking-long-term-state.md

```diff
@@ -0,0 +1,78 @@
+
+The current state differs from the ideal state in that the ideal state is much easier to reason about. In the ideal world, you test you changes in dev, and you promote them to prod, and everything is simple. In the current world, you have to test you *theoretical* changes in dev, and then *backfill* a similar type of change into the default VPC. The private subnets were backfilled into the default VPC in a manner such as this. Doing that required high level knowledge of terraform `imports` and `conditions`, whereas that knowledge wouldn't be required in the ideal state. In the ideal state, the networking is simply *the same, everywhere*.
+
+## Decisions
+
+There's two decisions to make here:
+
+- is the ideal state valuable enough to migrate to?
+- if so, how should we do that migration?
+
+### Decision 1 - Should We Migrate?
+
+This is a binary decision, should we migrate or not? Do our infrastructure engineers want to deal with a state, long term, where the action of promoting any networking change from `dev` to `prod` (ie. to "default") is complex? Do we still want to be dealing with that complexity 6 months, 2 years, 5 years from now? Given that we currently are a static site, and we are soon doing to have data, that makes this a "now or never" decision.
+
+Similarly, do our stakeholders will like name is a good time to incur some background infrastructure work for the sake of making our future lives easier?
+
+### Decision 2 - How Should We Migrate?
+
+There's functionally 2 good options here. The tradeoff is lead-time versus downtime. Specifically:
+
+#### 2a. Plan The Load Balancer And DNS Change On The Same Day
+
+Nava and MicroHealth coordinate a day where the new prod VPC load balancer will be created. Nava lets MicroHealth know that *at some point in that day* Nava will have a new load balancer URL. MicroHealth is responsible for receiving and deploying the load balancer URL on the same day, thus minimizing downtime. This should be able to happen during the course of a 90 minute meeting, if possible. That would limit the downtime to that same 90 minute window.
+
+This option:
+
+- Is an efficient use of both Nava and MicroHealth's time, but requires pre-planning.
+- Has a rather long downtime window of about 90 minutes max.
+
+#### 2b. Prepare A New Prod Load Balancer, Wait For DNS Changeover
+
+Nava prepares a new production load balancer URL inside the prod VPC, by deploying a new totally new production application. This application needs a name, lets say prod-beta. prod-beta can be deployed at any time, then MicroHealth is free to update the DNS record on their own time.
+
+- Requires the most prep work from Nava, and the new application would need to be permanently named prod-beta (or whatever we name it). Other name suggestions follow:
+  - prod-alpha (It just can't be called `prod`)
+  - live
+  - prod-0
```

Comment:
An idea just occurred to me. We could call this new prod `prod-east`, with an eye to eventually making a backup `prod-west`

?